### PR TITLE
[codex] Recognize subscription LLM profiles as configured

### LINKED ISSUE #1488: Treat ChatGPT subscription LLM profiles as configured without API keys
## Problem
Agent Canvas treats usable ChatGPT subscription-backed LLM profiles as unconfigured because the profile list response only exposes `api_key_set`, and subscription profiles intentionally do not have an API key.

## Goal
Allow local active LLM profiles that use ChatGPT subscription auth to pass the new-chat readiness check.

## Scope
- Fetch active profile detail when a local active profile lacks an API key.
- Treat subscription-auth profile detail as configured.
- Preserve the existing fast path for API-key-backed profiles.
- Add regression coverage for `gpt-5.5-sub`-style profiles.

## Acceptance criteria
- New-chat readiness succeeds for active subscription-backed profiles.
- API-key profiles keep the current behavior and performance path.
- Tests cover the subscription profile readiness case.

## Related PR
- OpenHands/agent-canvas#1458: https://github.com/OpenHands/agent-canvas/pull/1458

_This issue was updated by an AI agent (OpenHands) on behalf of Graham Neubig._

### PR DESCRIPTION
**Pull request by neubig**
HUMAN:

Manually verified that an active ChatGPT subscription profile (gpt-5.5-sub, no API key) now starts a new chat instead of showing the "LLM not configured" banner.

- [x] A human has tested these changes.

AGENT:

Subscription-auth profiles (e.g. ChatGPT subscription) are now recognized as configured. The fix is minimal and well-targeted, with unit tests covering the subscription path, the API-key fast path, and the no-key non-subscription negative case. End-to-end evidence (before/after screenshots) is included below.

---

## Why

The new-chat LLM readiness hook used the `/api/profiles` list response and only checked `activeProfile.api_key_set`. ChatGPT subscription profiles intentionally have no API key, and the list response does not include `auth_type`, so the frontend treated a usable subscription-backed profile as missing an LLM. This blocked a user with an active subscription profile (e.g. `gpt-5.5-sub`) from starting a new chat even though their profile was valid. Linked to issue #1488 / Linear APP-2495.

## Summary

- Treat local active LLM profiles as configured when their profile detail uses ChatGPT subscription auth, even when `api_key_set` is false.
- Fetch active profile detail only for local active profiles without an API key, preserving the existing fast path for API-key profiles.
- Add regression tests for the new-chat readiness hook: the subscription path, the API-key fast path (no detail fetch), and the no-key non-subscription negative case.

## Issue Number

#1488 (Linear APP-2495)

## How to Test

1. `npm install`
2. `npm test -- src/hooks/use-llm-configured.test.tsx` — all three cases pass.
3. `npm run typecheck && npm run build`
4. `npx eslint src/hooks/use-llm-configured.ts src/hooks/use-llm-configured.test.tsx`
5. Manually: with an active ChatGPT subscription LLM profile (`openhands/gpt-5.5-sub`, `auth_type: "subscription"`, no API key), open the home screen — the new-chat gate passes and no "LLM not configured" banner appears. Before the fix, the same profile showed the "Your LLM isn't set up yet" banner and blocked new chat.

## Video/Screenshots

Before/after screenshots captured locally with a mocked active ChatGPT subscription LLM profile (`openhands/gpt-5.5-sub`, `auth_type: "subscription"`, no API key) — the exact scenario from issue #1488.

**BEFORE — `main` (buggy):** an active subscription profile with no API key is treated as unconfigured. The new-chat readiness gate blocks and the home screen shows the "Your LLM isn't set up yet, so conversations won't run. Finish setup to get started." banner with a "Set up LLM" action.

before — main blocks new chat for an active subscription profile

**AFTER — this PR branch:** the same active subscription profile is recognized as configured (subscription auth), so the new-chat gate passes and the home screen is ready to start a conversation — no "LLM not configured" banner.

after — PR branch recognizes subscription profile as configured

## Type

- [x] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Breaking change
- [ ] Docs / chore

## Notes

The `.pr/` screenshot assets were committed to this branch (commit db7214de) and referenced via `raw.githubusercontent.com` URLs. The welcome onboarding modal and the @openhands/telemetry privacy/telemetry consent dialogs were dismissed before capturing, so only the new-chat readiness state is visible. A follow-up merge of `main` (commit 4a060c6a) restored the mock-LLM e2e comment script that CI (run from main) requires.

---
**🐳 Docker images for this PR**

• **GHCR package:** https://github.com/OpenHands/agent-canvas/pkgs/container/agent-canvas

| Component | Value |
|---|---|
| **Image** | `ghcr.io/openhands/agent-canvas` |
| **Architectures** | amd64, arm64 |
| **Agent Server** | `ghcr.io/openhands/agent-server:1.29.0-python` |
| **Automation** | `openhands-automation==1.0.0a12` |
| **Commit** | `4a060c6a26ed7911e0bca667159ed7bdabe79ce0` |

**Pull (multi-arch manifest)**
```bash
# Multi-arch manifest — Docker automatically pulls the correct architecture
docker pull ghcr.io/openhands/agent-canvas:sha-4a060c6
```

**Run**
```bash
docker run -it --rm \
  -p 8000:8000 \
  ghcr.io/openhands/agent-canvas:sha-4a060c6
```

**All tags pushed for this build**
```
ghcr.io/openhands/agent-canvas:sha-4a060c6-amd64
ghcr.io/openhands/agent-canvas:codex-subscription-profile-llm-configured-amd64
ghcr.io/openhands/agent-canvas:pr-1458-amd64
ghcr.io/openhands/agent-canvas:sha-4a060c6-arm64
ghcr.io/openhands/agent-canvas:codex-subscription-profile-llm-configured-arm64
ghcr.io/openhands/agent-canvas:pr-1458-arm64
ghcr.io/openhands/agent-canvas:sha-4a060c6
ghcr.io/openhands/agent-canvas:codex-subscription-profile-llm-configured
ghcr.io/openhands/agent-canvas:pr-1458
```

**About Multi-Architecture Support**
- Each tag (e.g., `sha-4a060c6`) is a **multi-arch manifest** supporting both **amd64** and **arm64**
- Docker automatically pulls the correct architecture for your platform
- Individual architecture tags (e.g., `sha-4a060c6-amd64`) are also available if needed

Closes #1488

### CODE REVIEW

File: src/hooks/use-llm-configured.test.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    } as Awaited<ReturnType<typeof OptionService.getConfig>>);
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+    setActiveSelection(null);
+    setRegisteredBackends([]);
+  });
+
+  it("treats an active subscription LLM profile without an API key as configured", async () => {
```
[the remark this answers is not in the document]
Comment (neubig) [reply src/hooks/use-llm-configured.test.tsx:94]: Added both cases in commit 3198fe01:

1. API-key profile (api_key_set: true) → isConfigured: true, and getProfile is NOT called (fast path intact).
2. No-key, non-subscription local profile (api_key_set: false, no auth_type: "subscription") → isConfigured: false, and getProfile IS called (detail fetch gated on key-less local profile).

All three tests in use-llm-configured.test.tsx pass locally.

**Comment by neubig** [comment]
Addressed the review feedback in 3198fe01 — added the two missing test cases (API-key fast-path regression and no-key non-subscription negative case). All three tests in use-llm-configured.test.tsx pass locally. Ready for another look.

**Comment by neubig** [comment]
Merged main into the branch (commit 4a060c6a) to restore the mock-llm e2e comment script that CI (run from main) requires. No logic changes to the fix. Ready for another look.
